intro & outro cues blog post

Be-ing opened the pull request:
This is long, but it needs to be because this is complicated.

Be-ing committed:
intro & outro cues blog post

Holzhaus commented:
Thanks. 

I'd find it helpful to put each sentence in a separate line which makes reviewing and commenting on the diff easier. The HTML generator will not insert a manual break unless there is a blank line between them so it should not make a difference visually.

This post sticks out a bit because it uses first person singular instead of first person plural or second person like the other  posts. I tried to avoid that in my posts, but if you want to keep it that way that'd be okay for me, too.

I'll try to review this properly tomorrow.

Holzhaus commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,91 @@
+<!-- screenshot -->
+
+It is up to you to decide where to place the intro end and outro start points.
+You can easily jump to the intro end and outro start points like hotcues, but that is not their primary purpose.
```
> You can easily jump to the intro end and outro start points like hotcues

How?

Holzhaus commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,91 @@
+This way I would know that the energy of the new track would pick up exactly as the old track ends.
+Depending on the musical content of each track, I might or might not bring the fader up on the new track before the last 1 or 2 seconds of the old track.
+
+Before Mixxx 2.3, I could mark the end of the intro with a hotcue, but I still had to jump to that hotcue and look at what time the deck was at to know how long the intro was.
```
OT: it would be really neat if Mixxx could display the seconds/beats/bars until the next hotcue.

Be-ing replied:
Yes that will be part of @hacksdump's project.

Holzhaus commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,91 @@
+However, it is good enough to give a human DJ a break without a major disruption to the mix.
+
+When there isn't a pandemic sweeping the globe, every week I DJ at a little gathering in my friend's yard while my friends spin fire.
+With the new AutoDJ modes, I can put two or three tracks in the AutoDJ queue with intro & outro sections marked and confidently step away from my gear for a few minutes so I can spin fire.
```
Same ending in two consecutive sentences.

Be-ing replied:
So what?

xeruf replied:
It does confuse my brain a little tbh ^^

Holzhaus commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,91 @@
+
+If I do not have the outro start point marked when I load a track, before I mix in the track, I skip ahead to the end of the track in my headphones to find the outro start point, then seek back to the intro start to get the track ready.
+If I don't have time to find the outro end point before mixing in the track, I can use the new deck cloning feature to drag and drop to a new deck, then seek ahead on the new deck and listen in my headphones to mark the outro start point.
+If I don't have time to do that either, I right click on the overview waveform where I think the outro starts and look how far that is from the end of the track to compare it to the length of the intro of the next track.
```
All three sent nces start with "If I", but that's intended here, right?

Be-ing replied:
Yeah I don't see a problem here.

Holzhaus commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,91 @@
+When there isn't a pandemic sweeping the globe, every week I DJ at a little gathering in my friend's yard while my friends spin fire.
+With the new AutoDJ modes, I can put two or three tracks in the AutoDJ queue with intro & outro sections marked and confidently step away from my gear for a few minutes so I can spin fire.
+It works well enough that no one even notices that the track changes but I'm not at my computer.
+
```
Do we want to add a little "get involved" section here? Maybe tease future plans for track sections (GSoC) or AutoDJ and encourage users to help testing or something like that?

Be-ing replied:
Yes thanks for the reminder.

Be-ing commented:
> This post sticks out a bit because it uses first person singular instead of first person plural or second person like the other posts. I tried to avoid that in my posts, but if you want to keep it that way that'd be okay for me, too.

I went into a lot of detail about my particular workflow, so I think it would be awkward to try write it differently. I think it makes it more relatable too.

foss- commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,107 @@
+Before we implemented these intro & outro cues, I calculated how long these sections were in my head.
+After I identified which track I wanted to mix in next, I would find the point in that track where I would want to have the old track completely faded out.
+Then, I would look at what time that was on the deck and keep that number in my memory.
+I would watch the how much time was remaining on the old track and listen for the time to start mixing out the old track.
```
I would watch ~~the ~~how much time was remaining on the old track and listen for the time to start mixing out the old track.

foss- commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,107 @@
+Then, I would look at what time that was on the deck and keep that number in my memory.
+I would watch the how much time was remaining on the old track and listen for the time to start mixing out the old track.
+
+For example, if I knew the old track had a fairly sudden ending with only 1 or 2 seconds without a lot going on at the end, and the next track had a 20 second intro, to avoid letting the energy of the mix drop if I let only that intro play, I could start the next track when the old track's deck showed 20 seconds remaining.
```
For example, if the old track had a fairly sudden ending with only 1 or 2 seconds and the next track had a 20 second intro with lower energy, to avoid dropping the energy of the mix by only having that low energy intro play, I would start the next track when the old track's deck showed 20 seconds remaining.

foss- commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,107 @@
+I would watch the how much time was remaining on the old track and listen for the time to start mixing out the old track.
+
+For example, if I knew the old track had a fairly sudden ending with only 1 or 2 seconds without a lot going on at the end, and the next track had a 20 second intro, to avoid letting the energy of the mix drop if I let only that intro play, I could start the next track when the old track's deck showed 20 seconds remaining.
+This would align the end of the old track with the end of the intro of the new track.
```
```suggestion
This would align the end of the old track with the end of the intro of the new track so that the energy of the new track would pick up right as the old track ends.
```

foss- commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,107 @@
+Although Mixxx could remember the hotcue, I'd still have to go to it every time I loaded the track to remember what position it was at.
+And what if the old track ended with a few seconds of silence or inaudibly quiet sounds?
+My calculation for when to press play on the next track would be thrown off.
+Marking the intro and outro *sections* rather than single points makes both of these much easier.
```
```suggestion
Marking the intro and outro *sections* rather than single points makes both situations much easier.
```

foss- commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,107 @@
+Now I can watch the time remaining on the old track and press play when it is equal to the length of the intro of the next track.
+
+If the old track has 1 or 2 seconds of silence at the end, the analyzer will mark the outro end where the sound becomes too quiet to hear.
+In this case, instead of watching the time remaining on the deck, I can hover my mouse cursor over the outro end line on the overview waveform and Mixxx will show me how much time is left until that point.
```
```suggestion
add screenshot showing that scenario?
```

foss- commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,107 @@
+Now I can watch the time remaining on the old track and press play when it is equal to the length of the intro of the next track.
+
+If the old track has 1 or 2 seconds of silence at the end, the analyzer will mark the outro end where the sound becomes too quiet to hear.
+In this case, instead of watching the time remaining on the deck, I can hover my mouse cursor over the outro end line on the overview waveform and Mixxx will show me how much time is left until that point.
```
```suggestion
add screenshot showing that scenario?
```

Holzhaus replied:
I suggest to leave the text as-is and add an animated gif (e.g. using [Peek](https://github.com/phw/peek))

Holzhaus commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -15,13 +15,15 @@
 The analyzer places the intro start point at the first sound and the outro end point at the last sound. The first and last sound are determined by the first time the signal rises above -60 dBFS and the last time it goes below -60 dBFS.
-<!-- screenshot -->
+
+<img src="/static/images/news/intro-outro/intro-start-outro-end.png" style="max-width: 465px;" alt="A track with the automatically placed intro start and outro end cues"/>
```
```suggestion
![A track with the automatically placed intro start and outro end cues](/static/images/news/intro-outro/intro-start-outro-end.png) 
```

Be-ing replied:
I think we need CSS for this. It seems to be working with 1e6b3d4 when making my browser window very narrow.

Holzhaus replied:
I'd rather not mix Markdown and HTML. Can we try to use plain markdown and handle the image resizing in CSS automatically?

This should work. It doesn't stretch small images to the full page width (it centers them instead), but large images are downscaled if they wouldn't fit:

```css
img {
  /* Center the image horizontally */
  display:block;
  margin-left:auto;
  margin-right:auto;
  /* Automatically downscale image to fit in parent container */
  max-width: 100%;
  height: auto;
}
```

If we can, we should use markdown only. That would also make the transition easier if we want to switch away from cactus (e.g. to Hugo or something like that).

Holzhaus replied:
Check #93

Holzhaus reviewed (commented):
The images are cut off on my phone. Let's check if it works with plain markdown image syntax.

ronso0 commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,141 @@
+<img src="/static/images/news/intro-outro/intro-outro-full.png" style="max-height: 137px; max-width: 100%;" alt="A track with the full intro and outro marked"/>
+</a>
+
+## Intro & Outro Mixing Techniques Before Mixxx 2.3 ##
```
This section is way too long IMO. People do _not_ want to read in great length about what was _not_ possible before, they want to know about the new feature.
To point out the improvement, can you distill this and put like 1-3 sentences of it into the next section?

Be-ing replied:
I think it is helpful to establish the context for why these features were created. I'll try to edit this to make it shorter.

Be-ing replied:
On further though, you're right. I removed the whole section.

ronso0 commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,141 @@
+
+<a href="/static/images/news/intro-outro/autodj-align-starts.png">
+<img src="/static/images/news/intro-outro/autodj-align-starts.png" style="max-width:100%;" alt="AutoDJ aligning the start of the intro and outro"/>
+</a>
```
Here and below: please include the AutoDJ controls row in the screenshot so that the images are easier to relate to the transition modes.

xeruf commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,141 @@
+</a>
+
+It is up to you to decide where to place the intro end and outro start points.
+You can easily jump to the intro end and outro start points by pressing the buttons like hotcues, but that is not their primary purpose.
```
I think this line should be separated from the previous one, they aren't really coherent.

xeruf replied:
Also, you can jump to any of these points by pressing the respective buttons, not only "intro end and outro start"

Be-ing replied:
I simply removed that sentence.

xeruf commented on pages/news/2020-05-16-intro-outro-sections.html:
```diff
@@ -0,0 +1,141 @@
+
+## Intro & Outro Mixing Techniques Before Mixxx 2.3 ##
+
+Before we implemented these intro & outro cues, I calculated how long these sections were in my head.
```
I am wondering if you should use passive/neutral more often rather than first person pronouns, e.g.:
```suggestion
Before the implementation of intro & outro cues, I calculated how long these sections were in my head.
```
My gut tells me there's too much "I" in the text ^^

Be-ing replied:
I removed most uses of "I" except the short anecdote.

Be-ing commented:
I removed the section about what I had to do with old versions of Mixxx and removed the use of "I". The text and screenshots could be easily adapted for the manual now.

Be-ing commented:
I changed the images to use Markdown instead of HTML. The Netlify preview looks good on my phone. Ready for merge?

Holzhaus commented on pages/news/2020-07-08-intro-outro-sections.html:
```diff
@@ -0,0 +1,106 @@
+{% filter markdown:"extra" %}
+
+Do you use hotcues to mark the points you use to mix in and out of tracks? Have you wished AutoDJ knew to mix in and out at those points?
+In Mixxx 2.3, we have introduced new intro & outro cues for these purposes.
```
```suggestion
In Mixxx 2.3, we'll introduce new intro & outro cues for these purposes.
```

Or something similar, so that it doesn't sound like 2.3 has already been released.

Holzhaus commented on pages/news/2020-07-08-intro-outro-sections.html:
```diff
@@ -0,0 +1,106 @@
+
+![A track with the full intro and outro marked](/static/images/news/intro-outro/intro-outro-full.png){: style="max-height: 137px" }
+
+## Mixing With The Intro & Outro Cues ##
```
```suggestion
## Mixing With The Intro & Outro Cues
```

Holzhaus commented on pages/news/2020-07-08-intro-outro-sections.html:
```diff
@@ -0,0 +1,106 @@
+
+Here, the outro is approximately 30 seconds long. However, the analyzer placed the outro end point before the end of the track, so the outro is actually a little bit shorter than 30 seconds. Without marking the outro start point, you would have to do some math in your head to find exactly how long the outro is.
+
+## Intro & Outro Cues With AutoDJ ##
```
```suggestion
## Intro & Outro Cues With AutoDJ
```

Holzhaus commented on pages/news/2020-07-08-intro-outro-sections.html:
```diff
@@ -0,0 +1,106 @@
+If you still want the old behavior of AutoDJ using a fixed number of seconds to crossfade, that is still available with the "Full Track" mode.
+The new "Skip Silence" mode behaves the same way, but cuts out the silence at the beginning and end of tracks.
+
+## Section Detection ##
```
```suggestion
## Section Detection
```

Holzhaus commented on pages/news/2020-07-08-intro-outro-sections.html:
```diff
@@ -0,0 +1,106 @@
+
+## Section Detection ##
+
+Now that we know how useful it is to mark intro & outro sections, we are planning to take this concept further. As part of our [Google Summer of Code 2020 projects](/news/2020-05-05-mixxx-gsoc-projects-2020), we'll be adding downbeat, phrase, and section markers to Mixxx and extending the analyzers to detect these.
```
We could make it more personal and name the students:
```suggestion
Now that we know how useful it is to mark intro & outro sections, we are planning to take this concept further. As part of our [Google Summer of Code 2020 projects](/news/2020-05-05-mixxx-gsoc-projects-2020), our GSoC students Harshit and Cristiano will work on adding downbeat, phrase, and section markers to Mixxx and extending the analyzers to detect these.
```

Holzhaus commented on pages/news/2020-07-09-intro-outro-sections.html:
```diff
@@ -0,0 +1,106 @@
+title: Intro & Outro Cues, Silence Detection, and AutoDJ Improvements
```
The other posts use this as a prefix for the title:

```suggestion
title: New in 2.3: Intro & Outro Cues, Silence Detection, and AutoDJ Improvements
```

Be-ing replied:
done

Be-ing commented:
Ready?

Holzhaus reviewed (approved):
LGTM, thank you. Should the be merged now or later?

Be-ing commented:
Now
